packages/mc-scripts/src/commands/login.js

```diff
@@ -0,0 +1,98 @@
+const path = require('path');
+const { promisify } = require('util');
+const read = promisify(require('read'));
+const fetch = require('node-fetch');
+const fs = require('fs');
+const homedir = require('os').homedir();
+const { processConfig } = require('@commercetools-frontend/application-config');
+const { MC_API_URLS, DEFAULT_CREDENTIALS } = require('../constants');
+
+function validateEmail(email) {
+  const re = /\S+@\S+\.\S+/;
+  return re.test(email);
+}
+
+const getEmail = async () => {
+  let email = await read({ prompt: 'Email: ' });
+  if (!validateEmail(email)) {
+    console.log('Please input a valid email');
+    email = await getEmail();
+  }
+  return email;
+};
+
+const getPassword = async () => {
+  return await read({ prompt: 'Password: ', silent: true });
+};
+
+const getUpdatedCredentials = (oldCredentials, cloudIdentifier, token) => {
+  oldCredentials[cloudIdentifier] = token;
+  return JSON.stringify(oldCredentials);
+};
+
+const updateCredential = (cloudIdentifier, token) => {
```

Comment:
I think it would be nice to have a "storage" abstraction object, something like `storage.{get,set}`.
So all the logic to write/read from file is handled within that object/module.

Reply:
I know there are different ways of implementing such an abstraction but I wanted to leave here a proposal in case it may help:
```
const { existsSync, writeFileSync, readFileSync } = require('fs');

class CredentialsStorage {
  constructor(configFilePath) {
    this.configFilePath = configFilePath;
  }

  _updateConfigFile(configuration) {
    writeFileSync(this.configFilePath, JSON.stringify(configuration), { encoding: 'utf8' });
  }

  _getAllCredentials() {
    if (!existsSync(this.configFilePath)) {
      this._updateConfigFile({ credentials: [] });
    }
    const data = readFileSync(this.configFilePath, { encoding: 'utf8' });
    return JSON.parse(data);
  }

  getCredentials(cloudIdentifier) {
    const allCredentials = this._getAllCredentials();
    return allCredentials[cloudIdentifier];
  }

  storeCredentials(cloudIdentifier, token) {
    const allCredentials = this._getAllCredentials();
    allCredentials[cloudIdentifier] = token;
    this._updateConfigFile(allCredentials);
  }
}

export default CredentialsStorage;
```